### ISSUE TITLE
slurm test fails

### ISSUE THREAD

**Issue by ltalirz** [marvel-nccr/quantum-mobile#87]
For some reason, submitting the slurm test fails.
```
2019-01-08 00:36:10,687 p=78996 u=leopold |  TASK [marvel-nccr.slurm : do test submission] **********************************
2019-01-08 00:36:33,967 p=78996 u=leopold |  fatal: [default]: FAILED! => changed=true
  cmd:
  - ${HOME}/.tests/slurm/test-submit.sh
  delta: '0:00:22.719388'
  end: '2019-01-08 00:36:33.432585'
  msg: non-zero return code
  rc: 2
  start: '2019-01-08 00:36:10.713197'
  stderr: ''
  stderr_lines: []
  stdout: |-
    Writing to folder: /home/max/.tests/slurm/test1
    Submitted batch job 2
    Current queue status:
                 JOBID PARTITION                 NAME     USER    STATE       TIME TIME_LIMI  NODES NODELIST(REASON)
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
      [Waiting for job to finish...]
                     2      jobs       testsubmission      max  PENDING       0:00 UNLIMITED      1 (Resources)
    TIMEOUT while waiting for job to be scheduled!
  stdout_lines: <omitted>
2019-01-08 00:36:33,970 p=78996 u=leopold |     to retry, use: --limit @/Users/leopold/Personal/Postdoc-MARVEL/repos/quantum-mobile/playbook.retry
```

I guess it worked on docker, why is it failing here?

**Comment by ltalirz** [comment]
@giovannipizzi This is something I had already encountered - the slurm node is stuck in `drain` state

```
$ sinfo
PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
jobs*        up   infinite      1  drain qmobile
```

I already wrote down before how to get out of this state:
```
# in case the node is stuck in state "drain", recover via
# sudo -su slurm scontrol
# scontrol: update NodeName=qmobile State=DOWN Reason="undraining"
# scontrol: update NodeName=qmobile State=RESUME
```

I probably fixed it and believed it was fine then but it seems like it has come back (?).
Any ideas how to address this?

**Comment by ltalirz** [comment]
Interestingly, this problem does not seem to occur on travis.
The only difference in the ansible role is that on virtualbox this task is skipped:

```
- lineinfile:
    path: /etc/hosts
    regexp: "^(172\\.17\\.[0-9]+\\.[0-9]+)\\s+.*$"
    line: "\\1 {{ slurm_hostname }}"
    backrefs: true
    unsafe_writes: true
```
(`slurm_hostname` being `qmobile` on virtualbox)
Hm... is something like this necessary also on virtualbox?

This is the hosts file on virtualbox:
```
127.0.0.1       localhost
127.0.1.1 localhost

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
10.0.2.15 qmobile
```
@giovannipizzi Does this look fine to you?

**Comment by ltalirz** [comment]
> The skipped role would probably not do anything on virtualbox, I guess, as there is no IP of the form 172.17.. in the /etc/hosts file. Do you agree?

Sure.

> could you double check that 10.0.2.15 is indeed the correct IP address?
```
$ ifconfig 
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.2.15  netmask 255.255.255.0  broadcast 10.0.2.255
        inet6 fe80::a00:27ff:fec2:be11  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:c2:be:11  txqueuelen 1000  (Ethernet)
        RX packets 1991  bytes 2061598 (2.0 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 997  bytes 77543 (77.5 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 4853  bytes 1314372 (1.3 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 4853  bytes 1314372 (1.3 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

>  (is this hardcoded or automatic

It's automatic.

> do you know why the nodes go into the DRAIN state?

No

> Is there any explanation in the log files of why this happened?

There are indeed problems in `/var/log/slurm-llnl/slurmctld.log`

```
[2019-01-08T00:36:04.754] error: Configured MailProg is invalid
[2019-01-08T00:36:04.754] slurmctld version 17.11.2 started on cluster cluster
[2019-01-08T00:36:04.799] layouts: no layout to initialize
[2019-01-08T00:36:04.852] layouts: loading entities/relations information
[2019-01-08T00:36:04.852] error: Could not open node state file /var/lib/slurm-llnl/slurmctld/node_state: No such file or directory
[2019-01-08T00:36:04.852] error: NOTE: Trying backup state save file. Information may be lost!
[2019-01-08T00:36:04.852] No node state file (/var/lib/slurm-llnl/slurmctld/node_state.old) to recover
[2019-01-08T00:36:04.853] error: Could not open job state file /var/lib/slurm-llnl/slurmctld/job_state: No such file or directory
[2019-01-08T00:36:04.853] error: NOTE: Trying backup state save file. Jobs may be lost!
```
This is from `/var/log/slurm-llnl/slurmd.log`
```
[2019-01-08T00:36:05.659] error: Domain socket directory /var/spool/slurmd: No such file or directory
[2019-01-08T00:36:05.703] error: Node configuration differs from hardware: Procs=2:2(hw) Boards=1:1(hw) SocketsPerBoard=2:1(hw) CoresPerSocket=1:2(hw) ThreadsPerCore=1:1(hw)
[2019-01-08T00:36:05.703] Message aggregation disabled
[2019-01-08T00:36:05.703] CPU frequency setting not configured for this node
[2019-01-08T00:36:05.710] slurmd version 17.11.2 started
[2019-01-08T00:36:05.716] slurmd started on Tue, 08 Jan 2019 00:36:05 +0100
[2019-01-08T00:36:05.716] CPUs=2 Boards=1 Sockets=2 Cores=1 Threads=1 Memory=985 TmpDisk=63281 Uptime=1467 CPUSpecList=(null) FeaturesAvail=(null) FeaturesActive=(null)
```

Will look into the reasons for this if I find the time.

**Comment by ltalirz** [comment]
This issue should be fixed at marvel-nccr/ansible-role-slurm@5d278262912cc419165c61c9f00ef523ddadd1ea

I've tested: that sbatch <job> works fine, continues to work fine after a restart and even after powering off the machine in virtualbox and restarting.

Closing this issue - happy to reopen if the problem resurfaces.
